Add tracking attributes in share block

Ruchika4 opened the pull request:
* Add tracking attributes in share block for analytics

Resolves: [MWPW-134287](https://jira.corp.adobe.com/browse/MWPW-134287)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ruchika/share-analytics
- After: https://share-analytics--milo--ruchika4.hlx.page/drafts/ruchika/share-analytics

chrischrischris reviewed (approved):
Since you are working in share.js, can you also fix the eslint errors?

Ruchika4 commented:
> Since you are working in share.js, can you also fix the eslint errors?

I have fixed it now

Ruchika4 commented:
> ## [Codecov](https://app.codecov.io/gh/adobecom/milo/pull/1023?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom) Report
> > Merging [#1023](https://app.codecov.io/gh/adobecom/milo/pull/1023?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom) ([d9240e5](https://github.com/adobecom/milo/commit/d9240e52802a812698c0a387aa62bdb958c2c64a)) into [main](https://app.codecov.io/gh/adobecom/milo/commit/bf2059995bbe483ed9e2a2b844022ed9e3160029?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom) ([bf20599](https://github.com/adobecom/milo/commit/bf2059995bbe483ed9e2a2b844022ed9e3160029)) will **increase** coverage by `0.00%`.
> > The diff coverage is `95.94%`.
> 
> ```diff
> @@           Coverage Diff           @@
> ##             main    #1023   +/-   ##
> =======================================
>   Coverage   95.10%   95.10%           
> =======================================
>   Files         127      127           
>   Lines       31767    31814   +47     
> =======================================
> + Hits        30212    30257   +45     
> - Misses       1555     1557    +2     
> ```
> 
> [Files Changed](https://app.codecov.io/gh/adobecom/milo/pull/1023?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom)	Coverage Δ	
> [libs/blocks/share/share.js](https://app.codecov.io/gh/adobecom/milo/pull/1023?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom#diff-bGlicy9ibG9ja3Mvc2hhcmUvc2hhcmUuanM=)	`96.47% <95.94%> (-3.53%)`	⬇️
> ... and [3 files with indirect coverage changes](https://app.codecov.io/gh/adobecom/milo/pull/1023/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom)
> 
> 📣 We’re building smart automated test selection to slash your CI/CD build times. [Learn more](https://about.codecov.io/iterative-testing/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=adobecom)

I have added the unit test for code I added and codecov is catching few lines from code that existed before as not covered through test. I can add more unit test cases for those in a PR later.
